src/core/defs.jl

```diff
@@ -512,15 +512,14 @@ to restore values between trials.
 
 """
 function copy_external_params(md::ModelDef)
-    external_params = Dict{Symbol, ModelParameter}()
-
-    for (key, obj) in md.external_params
-        external_params[key] = obj isa ScalarModelParameter ? ScalarModelParameter(obj.value) : ArrayModelParameter(copy(obj.values), obj.dimensions)
-    end
-
+    external_params = Dict{Symbol, ModelParameter}(key => copy(obj) for (key, obj) in md.external_params)
     return external_params
 end
 
+Base.copy(obj::ScalarModelParameter{T}) where T = ScalarModelParameter{T}(copy(obj.value))
+
+Base.copy(obj::ArrayModelParameter{T})  where T = ArrayModelParameter{T}(copy(obj.values), obj.dimensions)
```

Comment:
Can't we use ``deepcopy`` for that and not add a method here? It seems to me that you are redefining ``copy`` here to do what ``deepcopy`` already does.

Reply:
There's no need to make a copy of the the dimensions. It's a tradeoff between defining two one line functions and copying stuff we don't need to copy by relying on deepcopy. I could go either way.